src/fwrite.c

```diff
@@ -198,6 +199,9 @@ void writeFloat64(const void *col, int64_t row, char **pch)
     }
   } else if (x == 0.0) {
     *ch++ = '0';   // and we're done.  so much easier rather than passing back special cases
+    if (forceDecimal) {
```

Comment:
is there a way to use a flag like `needs_dec=false;` up-front to only add `*ch++=dec;` once just before returning?

Reply:
But iinm we write backwards into the buffer, so adding at the end will place it after the whole field, not the last digit